### PR TITLE
chain-rest: upgrade get latest txs

### LINKED ISSUE #596: Implement better REST latest txs handler
The code should try to binary search first to try to find the block. Talk to @sorawit for details.

### PR DESCRIPTION
**Pull request by evilpeach**
Fixes #596 

[Example1](http://rpc.alpha-debug.bandchain.org:1317/d3n/txs/latest?page=1&limit=100)
[Example2](http://scan-alpha.surge.sh/)

### CODE REVIEW

**Comment by sorawit** [comment]
This is not binary search implementation, but let's start with this first and see how it goes. At least it should now not break. It might get slow if we have LOTS of txs and we will figure that out later. Binary search can be used to find the highest height with enough txs and reuse pretty much the same code for finding txs.

File: chain/client/client.go [hunk 1]
```diff
@@ -90,26 +90,52 @@ func LatestTxsRequestHandlerFn(cliCtx context.CLIContext) http.HandlerFunc {
...
-		for ; startIdx >= endIdx && startIdx >= 0; startIdx-- {
-			result.Txs = append(result.Txs, searchResult.Txs[startIdx])
+
+		if startIndex < 1 {
+			startIndex = 1
 		}
+
+		if endIndex < 1 {
+			result.Count = 0
+			rest.PostProcessResponseBare(w, cliCtx, result)
```
Comment (sorawit) [inline chain/client/client.go:117]: I think page total should also be in here. Maybe set it above?

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/client/client.go [hunk 2]
```diff
@@ -90,26 +90,52 @@ func LatestTxsRequestHandlerFn(cliCtx context.CLIContext) http.HandlerFunc {
...
+			return
+		}
+
+		todoIndex := startIndex
+		for todoIndex <= endIndex {
+			pageOfTodoIndex := (todoIndex-1)/limit + 1
```
Comment (taobun) [inline chain/client/client.go:123]: If `limit` is more than 100, it leads to infinite loop.

**Review by taobun — CHANGES_REQUESTED** [review]

File: chain/client/client.go [hunk 3]
```diff
@@ -85,31 +85,62 @@ func LatestTxsRequestHandlerFn(cliCtx context.CLIContext) http.HandlerFunc {
 			return
 		}
 
+		// limit maximum must be 100.
+		if limit > 100 {
```
Comment (taobun) [inline chain/client/client.go:88]: Return error

**Review by taobun — CHANGES_REQUESTED** [review]

File: chain/client/client.go [hunk 4]
```diff
@@ -85,31 +85,63 @@ func LatestTxsRequestHandlerFn(cliCtx context.CLIContext) http.HandlerFunc {
...
+			rest.WriteErrorResponse(w, http.StatusBadRequest, "limit must be less than or equal 100.")
```
Comment (sorawit) [inline chain/client/client.go]: must not be greater than

**Review by sorawit — CHANGES_REQUESTED** [review]

**Review by taobun — APPROVED** [review]

**Review by sorawit — APPROVED** [review]

**Pull request merged**